src/request.rs

```diff
@@ -44,7 +57,33 @@ impl fmt::Display for Request {
             let values = values.join(",");
             writeln!(f, "{}: {}", name, values)?;
         }
-        writeln!(f, "{}", String::from_utf8_lossy(&self.body))
+
+        match self.body_print_limit {
+            BodyPrintLimit::Limited(limit) if self.body.len() > limit => {
+                // We need to use from_utf8_lossy because the limit may land within a utf8
+                // character.
+                let truncated = String::from_utf8_lossy(&self.body[..limit]);
```

Comment:
I'm unsure about this—we need `lossy` because we might be asking for a slice that has non-valid terminal utf8 character, but `lossy` will also mask any other invalid characters that might occur before.

Ideally, we want to separate "print at most N bytes" from "print binary-like bodies" in terms of concerns (and configuration options).
I'd suggest we try to decode the entire body in a non-lossy way before trying to print it. If it decodes correctly, then we need to find a way to print up to N bytes by picking a "good" terminal index—either by printing a bit less than N or a bit more, it doesn't matter much.

Wdyt?

Reply:
I did consider growing or shrinking the attempted decode first but opted for simplicity first, the maximum number of bytes for a character is 4 so if we fall within a boundary we only need to try 3 more decodes max before we can assume it's a binary body. 

Going back to look at the code before sending I have a feeling my mindset was "unlimited binary body printing bad, but if there's a print size limit it's not as bad". But I guess for consistency we can just never print a binary body because it's kinda useless.

You might want to make an issue for saving some serde encoded version of the requests to file if they aren't being printed (if wiremock doesn't do this already). Could be useful for the ol' binary body test failure debugability

Reply:
Let's add the code required to be on a "good split" and then I think we're good to go for merging 👍🏻 
I'll open an issue for storing the responses on disk.

Reply:
Okay added, I guess the next question is where are the tests so I can put a bit of effort into making sure  it all works as expected.

Reply:
We don't, but I think we can be satisfied by extracting the limit-searching bit into its function and unit testing that one.

Reply:
Will do